volatility/framework/plugins/windows/malfind.py

```diff
@@ -101,11 +105,24 @@ def list_injections(
             if (vad.get_private_memory() == 1
                     and vad.get_tag() == "VadS") or (vad.get_private_memory() == 0
                                                      and protection_string != "PAGE_EXECUTE_WRITECOPY"):
-                if cls.is_vad_empty(proc_layer, vad):
+                if self.is_vad_empty(proc_layer, vad):
                     continue
 
                 data = proc_layer.read(vad.get_start(), 64, pad = True)
-                yield vad, data
+                if output_dir is not None:
+                    try:
+                        filedata = interfaces.plugins.FileInterface("pid.{0}.{1:#x}.dmp".format(
+                            proc.UniqueProcessId, vad.get_start()))
+                        a = proc_layer.read(vad.get_start(), vad.get_end() - vad.get_start(), pad=True)
+                        filedata.data.write(a)
+
+                        self.produce_file(filedata)
+                        result_text = "Stored {}".format(filedata.preferred_filename)
+                    except Exception as ex:
+                        print(ex)
+                        result_text = "Unable to dump memory region at {0:#x}".format(vad.get_start())
+
+                yield vad, data, result_text
```

Comment:
I had previously been providing `result_text`, but I realized there's not a great deal of information conveyed and we should probably do this using `vollog` (again, examples in #290).  If this is just dumping a vad, you may be able to use the classmethod from vaddump (or as of #290, from vadinfo)...